addons/project/static/src/js/portal_project.js

```diff
@@ -0,0 +1,50 @@
+odoo.define('web.web_client', function (require) {
+    var WebClient = require('web.WebClient');
+    var web_client = new WebClient();
+    return web_client;
+});
+
+odoo.define('project.portal_project', function(require) {
+    require('web.dom_ready');
+
+    var $target = $('portal-project');
+
+    if (!$target.length) {
+        return $.Deferred().reject("DOM doesn't contain 'portal-project'");
+    };
+    if (!$target.data('params')) {
+        return $.Deferred().reject("No params were passed to 'portal-project'")
+    };
+
+    var params = $target.data('params');
+    switch (params['viewType']) {
+        case 'kanban':
+            var PortalKanban = require('project.PortalKanban');
+            var portalKanban = new PortalKanban(params);
+            portalKanban.appendTo($target);
+            break;
+        case 'form':
+            var PortalForm = require('project.PortalForm');
+            var portalForm = new PortalForm(params);
+            portalForm.appendTo($target);
+            break;
+        default:
+            return $.Deferred().reject("No view type was specified in the params passed to 'portal-project'");
+    };
+
+    setInterval(function() {
+        resizeIFrame($(document).height())
+    }, 100);
```

Comment:
In general we try to avoid using intervals or timeout in general.
also, is it really necessary to resize the iframe every 0.1 sec ?
there are other ways to wait for something to be ready before executing other instructions (callback, promise or.. see with js team, they are better than me on this :D)

Reply:
I'll see if anyone at the JS team has a better suggestion. I also didn't like writing this ;-)
Also, 0.1s delay is maybe a bit excessive indeed, I just wanted to make it seamless :-)

Reply:
Well that did create a problem with the mobile version but I fixed that as well. `document.body` didn't take the kanban cards' heights into account as their position was absolute. My (hacky, I admit) fix has been to set their position to relative if active and absolute if not.

```javascript
function adjustPositions() {
    $('.o_kanban_group').each(function () {
        if ($('.o_current').attr('data-id') === $(this).attr('data-id')) {
            $(this).css('position', 'relative');
        } else {
            $(this).css('position', 'absolute');
        };
    });
};
```
and
```css
@media (max-width: 767.98px) {
    .o_kanban_view.o_kanban_grouped[class] .o_kanban_group:not(.o_column_folded) {
        top: 0;
    }
}
```